eng/scripts/Get-BinarySizes.ps1

```diff
@@ -0,0 +1,43 @@
+[CmdletBinding()]
+param(
+    [Parameter()]
+    [string] $BuildDirectory = "$PSScriptRoot../../../build/",
+
+    [Parameter()]
+    [string] $ServiceDirectory = "*",
+
+    [Parameter()]
+    [switch] $CI = ($null -ne $env:SYSTEM_TEAMPROJECTID),
+
+    [Parameter()]
+    [hashtable]$ExtraLabels = @{}
+)
+
+$searchPath = "$BuildDirectory/sdk/$ServiceDirectory/*/*.a"
+if ($IsWindows) {
+    $searchPath = "$BuildDirectory/sdk/$ServiceDirectory/*/*/*.lib"
+}
+
+$binaries = Get-ChildItem -Path $searchPath
+
+if ($CI) {
+    foreach ($binary in $binaries) {
+        $metricLogObject = @{
+            name = "BinarySize";
+            value = $binary.Length
+            timestamp = (Get-Date -AsUTC).ToString()
+            labels = @{
+                BinaryName = $binary.Name;
+            } + $ExtraLabels
+        }
+
+        $metricLogJson = ConvertTo-Json $metricLogObject -Depth 2 -Compress
```

Comment:
Is specifying Depth necessary?

Reply:
In this case, no. The default depth is 2 and today these objects do not go deeper than 2.